packages/bot-skeleton/src/scratch/blocks/Binary/Trade Definition/trade_definition_multiplier.js

```diff
@@ -80,6 +80,20 @@ Blockly.Blocks.trade_definition_multiplier = {
             const block_types_in_multiplier = [];
             blocks_in_multiplier.forEach(block => {
                 block_types_in_multiplier.push(block.type);
+                const block_multiplier_take_profit = block.childValueToCode('multiplier_take_profit', 'AMOUNT');
+                const block_multiplier_stop_loss = block.childValueToCode('multiplier_stop_loss', 'AMOUNT');
+                if (block_multiplier_take_profit <= 0 || block_multiplier_stop_loss <= 0) {
+                    block.setDisabled(true);
+                }
+
+                if (block.type === 'multiplier_stop_loss' && block_multiplier_stop_loss > 0) {
+                    block.setDisabled(false);
+                }
+
+                if (block.type === 'multiplier_take_profit' && block_multiplier_take_profit > 0) {
+                    block.setDisabled(false);
+                }
```

Comment:
Since both of the condition is doing the same action can these be written in the else block of the first condition?

Reply:
@shafin-deriv action happens in iterator, we don't want that the action touch both of blocks, same as we don't wan't touch any other block except of highlighted with condition + except any other block. There are different blocks being iterated. Maybe there's a specific suggestion for improvement that won't break the current logic, I'm always happy to consider it ☺️

Reply:
first condition has 2 types of blocks

Reply:
I was just wondering if something like this could be done to avoid multiple checks if unnecessary If not, then it's okay.

`if(block.type === 'multiplier_stop_loss' || block.type === 'multiplier_take_profit') {
    if (block_multiplier_take_profit > 0 || block_multiplier_stop_loss > 0) {
        block.setDisabled(false);
    } else {
        block.setDisabled(true);
    }
}`

Reply:
@shafin-deriv in you suggestion exist error:
`if (block_multiplier_take_profit > 0 || block_multiplier_stop_loss > 0) { block.setDisabled(false); } else { block.setDisabled(true); } }`
for example operator '||': block_multiplier_**take_profit** > 0, condition is true, we will call block.setDisabled(false); for block_multiplier_**stop_loss**, but we don't expect it, this is an error

Reply:
that is why code reduction here will create errors, we need to change the state of a block separately, independent of other blocks

Reply:
set will be done to wrong block in your case

Reply:
Sure @maryia-matskevich-deriv I was just asking if something like the suggestion could be done. If there is no way to reduce the multiple checks then it's okay.